slack status trigger

sandeshit opened the pull request:
## Changes

* Slack trigger for status change.

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] tests
- [x] permission checks (tests here too)
- [x] translations

sandeshit committed:
chore(model): make slack_progress_notification field blankable

sandeshit committed:
chore(slack): add client for slack not enabled condition

sandeshit committed:
feat(slack): trigger for status change slack message

tnagorra commented on apps/project/serializers.py:
```diff
@@ -739,6 +737,11 @@ def update(self, instance: Project, validated_data: dict[str, typing.Any]) -> Pr
+            transaction.on_commit(
+                lambda: chain(
+                    push_project_to_firebase.si(updated_project.pk),
+                    send_slack_message_for_project.si(project_id=updated_project.pk, action="status-change"),
```
What if push_project_to_firebase fails? Does it still get invoked?

sandeshit replied:
Theoretically no, it should not invoke safar dai. Not tested though.

tnagorra replied:
push_project_to_firebase can fail. So, let's keep that in mind.

sandeshit replied:
Okay Safar dai. The chain should stop if push_project_to_firebase fails.

tnagorra replied:
How does the chain know if push_project_to_firebase failed?

sandeshit replied:
safar dai, got it from here
https://docs.celeryq.dev/en/stable/userguide/canvas.html

tnagorra replied:
which section? also the tasks internally catch errors so i am not sure the errors are propagated to the chain

thenav56 committed:
feat(slack): remove slack bot name config

we can control this from slack app's configuration

sandeshit committed:
chore(model): add slack user id in admin panel.

This commit adds the `slack_user_id` field to the Django admin panel for the User model.
The field is now configured to be blankable but not nullable, which is the opposite of its
previous configuration.

tnagorra commented on apps/user/models.py:
```diff
@@ -43,7 +43,7 @@ class User(AbstractUser):
 
     objects: CustomUserManager = CustomUserManager()  # type: ignore[reportAssignmentType]
-    slack_user_id = models.CharField(max_length=20, null=True, blank=False)
+    slack_user_id = models.CharField(max_length=20, null=False, blank=True)
```
Any reason to change this?

sandeshit replied:
By setting null=False and blank=True, the field can be optional in forms (like admin) and saves empty string in database. I misinterpreted it last time safar dai and did the opposite

tnagorra committed:
feat(slack): handle failure states on project status changes

tnagorra committed:
chore(slack): update default config in docker and base test

- add merge migration
- update icon for withdrawn project
